### DISCUSSION TITLE
Mermaid configuation options in marp

### DISCUSSION THREAD

**Discussion by Kugeleis** [marp-team/marp#479] [Q&A] [answered]
I successfully integrated a marmaid sankey graph in my presentation:
```markdown
---
marp: true
---

# Sankey

<div class="mermaid">
  sankey-beta
  %% source,target,value
  Project,Database,10
  People,Database,20
  Measurements,Database,100
</div>

---
```

Now I want to add configuration to this graph `showValues: false` ([example in docs](https://mermaid.js.org/syntax/sankey.html#example))
but neither
```md
config:
  sankey:
    showValues: false
```
nor
```js
<script type="module">
  import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';
  mermaid.initialize({ startOnLoad: true,
  showValues: false
  });
</script>
```
work.  

How do apply options to a mermaid graph in marp?

**Comment by yhatt** [comment, marked as answer]
Marp has no official support for mermaid, but if you have already successful to render, you can set correct configuration in both cases

```markdown
# Sankey

<div class="mermaid">
  ---
  config:
    sankey:
      showValues: false
  ---
  sankey-beta
  %% source,target,value
  Project,Database,10
  People,Database,20
  Measurements,Database,100
</div>

<script type="module">
  import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@latest/dist/mermaid.esm.min.mjs';
  mermaid.initialize({ startOnLoad: true });
</script>
```

Mermaid setting is not a part of Marp so it's important to place the independent frontmatter inside the mermaid code.

or:

```markdown
<div class="mermaid">
  sankey-beta
  %% source,target,value
  Project,Database,10
  People,Database,20
  Measurements,Database,100
</div>

<script type="module">
  import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@latest/dist/mermaid.esm.min.mjs';
  mermaid.initialize({
    startOnLoad: true,
    sankey: {
      showValues: false
    }
  });
</script>
```

To set the setting via `mermaid.initialize()`, set the setting object as `sankey` field, as described at "Configuration" section of https://mermaid.js.org/syntax/sankey.html.

**Comment by soaple** [comment]
You can use [MarkSlides](https://www.markslides.ai/)(Free web editor supports Marp) to edit Mermaid like below.
<img width="1512" alt="스크린샷 2023-10-15 오후 5 34 11" src="https://github.com/marp-team/marp/assets/34570624/abd69aa8-2fa0-4297-9a53-a05d3fc30808">

![sample](https://github.com/marp-team/marp/assets/34570624/7689d840-d934-4e53-9452-e0f29188775f)
